Updated test matrix

asmogo opened the pull request:
### Description

Moved the test matrix from the GitHub workflow into a shell script. This will reduce pipeline complexity and enable running the tests on your local computer

-----

### Notes to the reviewers

<!-- In this section you can include notes directed to the reviewers, like explaining why some parts
of the PR were done in a specific way -->

-----

### Suggested [CHANGELOG](https://github.com/cashubtc/cdk/blob/main/CHANGELOG.md) Updates

<!-- Please do not edit the actual changelog but note what you changed here. -->

#### CHANGED

#### ADDED

#### REMOVED

#### FIXED

----

### Checklist

* [ ] I followed the [code style guidelines](https://github.com/cashubtc/cdk/blob/main/CODE_STYLE.md)
* [ ] I ran `just final-check` before committing

asmogo commented on .github/workflows/ci.yml:
```diff
@@ -32,314 +42,102 @@ jobs:
-      - name: Build
-        run: nix develop -i -L .#msrv --command cargo build ${{ matrix.build-args }}
+      - name: Add WASM target
+        run: nix develop -i -L .#stable --command rustup target add wasm32-unknown-unknown
```
removed this here  https://github.com/cashubtc/cdk/pull/799/commits/d52fd75d354ff4a16b317d562c66510cb15c8f2a

thesimplekid commented:
Just approving to trigger ci. I have not gone through this in detail.

thesimplekid commented:
Sorry just realized this moved from draft. Should we go over it on the call tomorrow? I've added some nix caching to the ci so looks like there is a conflict we need to resolve now.

asmogo commented:
> Sorry just realized this moved from draft. Should we go over it on the call tomorrow? I've added some nix caching to the ci so looks like there is a conflict we need to resolve now.  

Oh no, we're good. Just moved the ticket out of draft for testing the approval. I will not be able to join the call tomorrow. Aiming for next week.

asmogo commented:
We’re running `cargo check —example` and `cargo run —example`. I’m not sure why we have this distinction.

I’ll add more changes to avoid repeating the same workflows.

We can speed up the integration test by moving some parts back to the ci.yaml file.
